File: Keywords/KeywordsCPlusPlus.java
///////////////////////////////////////////////////////////
//  KeywordsCPlusPlus.java
//  Implementation of the Class KeywordsCPlusPlus
//  Created on:      10-Nov-2015 20:34:05
//  Original author: udo_2
///////////////////////////////////////////////////////////

#if !defined(EA_D6842513_20D8_484a_B90D_A989215D270C__INCLUDED_)
#define EA_D6842513_20D8_484a_B90D_A989215D270C__INCLUDED_

#include "Parser.h"

class KeywordsCPlusPlus : public Parser
{

public:
	KeywordsCPlusPlus();
	virtual ~KeywordsCPlusPlus();

	hash_map Get();
	vector<string> GetKeywords();
	hash_map Read();
	void SetKeywords(vector<string> newVal);

private:
	vector<string> Keywords;

	bool Parse();

};
#endif // !defined(EA_D6842513_20D8_484a_B90D_A989215D270C__INCLUDED_)

File: Sources/Parsing/KeywordsCPlusPlus.cpp


///////////////////////////////////////////////////////////
//  KeywordsCPlusPlus.cpp
//  Implementation of the Class KeywordsCPlusPlus
//  Created on:      10-Nov-2015 20:34:05
//  Original author: udo_2
///////////////////////////////////////////////////////////


#include <QString>
#include "KeywordsCPlusPlus.h"
#include "Utilities.h"
#include "GeneratorCPlusPlus.h"
#include "ParserimplementationText.h"
#include "Base/BaseGenerator.h"
#include "../Errorhandling/Exceptionhandling.h"
#include "Base/BaseParserImplementation.h"
#include "iostream"

namespace NParser
{


using namespace std;
using namespace NGenerator;
#include "memory"


void  KeywordsCPlusPlus::registerObservers(NGenerator::BaseGenerator *observer)
{
    id = General::ParserId::Id::Keyword;
    parserObservers.push_back(&*observer);
}


void KeywordsCPlusPlus::Parse(const QString str)
{
    std::unique_ptr<ParserImplText> parser (new ParserImplText());
    try
    {
        vecValues = parser->doParseForVec(str);
    }
    catch(Errorhandling::Exceptionhandling &error)
    {
        // @todo:: log
        std::cout << error.whatDescription();
    }
    for(unsigned int i=0; i < parserObservers.size(); i++){
        parserObservers[i]->notifyDatasGenerated(id);
    }
}


void KeywordsCPlusPlus::giveData() const
{
   for(unsigned int i=0; i < parserObservers.size(); i++)
   {
       parserObservers[i]->receiveData(vecValues,id);
   }
}

}









